src/clover/analysis.py

```diff
@@ -104,10 +106,12 @@ def get_key_results(
     )
 
     # Compute the grid results.
-    if grid_input_profile is not None:
-        key_results.grid_daily_hours = np.sum(
-            grid_input_profile[: num_years * HOURS_PER_YEAR], axis=0
-        ) / (365 * num_years)
+    key_results.grid_daily_hours = collections.defaultdict(float)
```

Comment:
@BenWinchester error has been spotted here because first we don't have a for loop and second we don't have grid_energy working properly. Only getting EDL weirdly (suspected error in storage.py)